src/Core/Foundation/PrestaShopVersion/PrestaShopVersion.php

```diff
@@ -0,0 +1,214 @@
+    }
+
+    /**
+     * Returns if the current PrestaShop version is less than the provided version
+     *
+     * @param $version Must be a valid PrestaShop version string, for example "1.7.4.0"
+     *
+     * @return bool
+     *
+     * @throws InvalidArgumentException If the provided version is invalid
+     */
+    public function isLessThan($version)
+    {
+        return $this->versionCompare($version, '<');
+    }
+
+    /**
+     * Returns if the current PrestaShop version is less than or equal to the provided version
+     *
+     * @param $version Must be a valid PrestaShop version string, for example "1.7.4.0"
+     *
+     * @return bool
+     *
+     * @throws InvalidArgumentException If the provided version is invalid
+     */
+    public function isLessThanOrEqualTo($version)
+    {
+        return $this->versionCompare($version, '<=');
+    }
+
+    /**
+     * Returns if the current PrestaShop version is equal to the provided version
+     *
+     * @param $version Must be a valid PrestaShop version string, for example "1.7.4.0"
+     *
+     * @return bool
+     *
+     * @throws InvalidArgumentException If the provided version is invalid
+     */
+    public function isEqualTo($version)
+    {
+        return $this->versionCompare($version, '=');
+    }
+
+    /**
+     * Returns if the current PrestaShop version is not equal to the provided version
+     *
+     * @param $version Must be a valid PrestaShop version string, for example "1.7.4.0"
+     *
+     * @return bool
+     *
+     * @throws InvalidArgumentException If the provided version is invalid
+     */
+    public function isNotEqualTo($version)
+    {
+        return $this->versionCompare($version, '!=');
+    }
+
+    /**
+     * Compares the current PrestaShop version with the provided version depending on the provided operator
+     *
+     * @param $version Must be a valid PrestaShop version string, for example "1.7.4.0"
+     * @param $operator Operator for version_compare(), allowed values are: <, lt, <=, le, >, gt, >=, ge, ==, =, eq, !=, <>, ne
+     *
+     * @return boolean Result of the comparison
+     *
+     * @throws InvalidArgumentException If the provided version is invalid
+     */
+    private function versionCompare($version, $operator)
+    {
+        try {
+            $this->checkVersion($version);
+        } catch(InvalidVersionException $e) {
+            throw new InvalidArgumentException($e->getMessage());
+        }
+
+        return version_compare(AppKernel::VERSION, $version, $operator);
+    }
+
+    /**
+     * @param $version
+     * @throws InvalidVersionException If the provided version is invalid
+     */
+    private function checkVersion($version)
+    {
+        if (!is_string($version)) {
+            throw new InvalidVersionException('A valid version must be a string.');
+        }
+
+        $versionParts = explode('.', $version);
+
+        if (4 !== count($versionParts)) {
```

Comment:
Hum, you can replace both throw by a simple regex
```php
preg_match('~^(?:(\d+)\.?){1,}\d+$~', $version);
```
It's invalid if it don't match `1.N`, A version can be `1.0`, `1.9.1`, `1.2.3.4`. We can go further by adding support of alpha, beta, rc, etc. Making this class a little generic could be better for the community :)

Reply:
@PierreRambaud I will change it to a regex. - RC and Beta releases have the same version hardcoded like the stable release as far as I checked it (e.g. "1.7.3.0."). - Are you sure that it's good to make this class generic? I would keep it as strict as possible, if the string is d+.d+.d+.d+ it is valid, otherwise not. Why do we need 1.0 etc? I saw something like that hardcoded to ControllerTest.php but is there a strong reason to do that? - The method is private by the way

Reply:
@PierreRambaud I forget the most important: it is hard to accept "1.0" because version_compare() says it is less than "1.0.0". So it would be neceassary to handle all these cases seperatly which sounds bad. I changed it to a regex, please have a look. If a developer wants to compare "1.0" to the current PS version he has to transform it to "1.0.0.0" before - this does not seem too bad for me. Are you OK with this?

Reply:
Making this class working for modules can be a good idea I think, not only for "How prestashop managed their versions", wdyt @mickaelandrieu ?